apis/metal3.io/v1alpha1/hostfirmwaresettings_types.go

```diff
@@ -0,0 +1,80 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+type SettingsMap map[string]string
+
+type SchemaReference struct {
+	// `namespace` is the namespace of the where the schema is stored.
+	Namespace string `json:"namespace"`
+	// `name` is the reference to the schema.
+	Name string `json:"name"`
+}
+
+// HostFirmwareSettingsSpec defines the desired state of HostFirmwareSettings
+type HostFirmwareSettingsSpec struct {
+
+	// Settings are the desired firmware settings stored as name/value pairs.
+	// This will be populated with the actual firmware settings and only
+	// contain the settings that can be modified (i.e. not ReadOnly), to
+	// facilitate making changes.
+	// +patchStrategy=merge
+	Settings map[string]intstr.IntOrString `json:"settings" required:"true"`
```

Comment:
nit: we could think about declaring a type for this and naming it and SettingsMap in some way that makes it possible to distinguish them. Or maybe just ditch the SettingsMap type again since it's now used in only one place.

Reply:
I think that type will be useful and had added it in other patch. Updated.